internal/testutil/unit/http_mocker.go

```
@@ -23,14 +23,14 @@ const (
 )
 
 type MockHTTPDataConfig struct {
-	RunBeforeEach        func() error
-	RequestHandler       ManualRequestHandler
-	FilePathOverride     string
-	IsDiffSkipSuffixes   []string
-	IsDiffMustSubstrings []string
-	QueryVars            []string
-	AllowMissingRequests bool
-	AllowOutOfOrder      bool
+	RunBeforeEach        func() error         // Run by TestCase.PreCheck. Useful for reducing retry timeouts.
+	RequestHandler       ManualRequestHandler // Allow inspecting or overriding mocking behavior. Can be used to return 404 when a test has completed.
```

`Can be used to return 404 when a test has completed.`
can you give an example when this would be done?

---

[// Custom RequestHandler that runs after PlanModifier is done, to avoid `mock response not found`` errors in test cleanup functions](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/fc63160b29b16eb6b4adb3fdf2844a2434d3eecd/internal/testutil/unit/http_mocker_plan_checks.go#L117)